src/gui/TrackContentObjectView.cpp

```diff
@@ -918,12 +980,51 @@ void TrackContentObjectView::mouseReleaseEvent( QMouseEvent * me )
 	{
 		setSelected( !isSelected() );
 	}
-
-	if( m_action == Move || m_action == Resize || m_action == ResizeLeft )
+	else if( m_action == Move || m_action == Resize || m_action == ResizeLeft )
 	{
 		// TODO: Fix m_tco->setJournalling() consistency
 		m_tco->setJournalling( true );
 	}
+	else if( m_action == Split )
+	{
+		SampleTCO * leftTCO = dynamic_cast<SampleTCO*>( m_tco );
+
+		if (leftTCO)
+		{
+			leftTCO->setMarkerEnabled( false );
+
+			const int relativePixelPos = me->pos().x();
+			const float ppb = m_trackView->trackContainerView()->pixelsPerBar();
+			TimePos splitPos = relativePixelPos * TimePos::ticksPerBar() / ppb;
+
+			if ( !unquantizedModHeld(me) )
+			{
+				splitPos = quantizeMarkerPos( splitPos, me->modifiers() & Qt::ShiftModifier );
+			}
+
+			splitPos += m_initialTCOPos;
+
+			//Don't split if we slid off the TCO or if we're on the clip's start/end
+			//Cutting at exactly the start/end position would create a zero length
+			//clip (bad), and a clip the same length as the original one (pointless).
+			if ( splitPos > m_initialTCOPos && splitPos < m_initialTCOEnd )
+			{
+				leftTCO->getTrack()->addJournalCheckPoint();
+				leftTCO->getTrack()->saveJournallingState( false );
+
+				SampleTCO * rightTCO = new SampleTCO ( *leftTCO );
+
+				leftTCO->changeLength( splitPos - m_initialTCOPos );
+
+				rightTCO->movePosition( splitPos );
+				rightTCO->changeLength( m_initialTCOEnd - splitPos );
+				rightTCO->setStartTimeOffset( leftTCO->startTimeOffset() - leftTCO->length() );
+
+				leftTCO->getTrack()->restoreJournallingState();
+			}
+		}
+	}
+
```

Comment:
```suggestion
	else if (m_action == Split)
	{
		SampleTCO * leftTCO = dynamic_cast<SampleTCO*>(m_tco);

		if (leftTCO)
		{
			leftTCO->setMarkerEnabled(false);

			const int relativePixelPos = me->pos().x();
			const float ppb = m_trackView->trackContainerView()->pixelsPerBar();
			TimePos splitPos = relativePixelPos * TimePos::ticksPerBar() / ppb;

			if (!unquantizedModHeld(me))
			{
				splitPos = quantizeMarkerPos(splitPos, me->modifiers() & Qt::ShiftModifier);
			}

			splitPos += m_initialTCOPos;

			// Don't split if we slid off the TCO or if we're on the clip's start/end
			// Cutting at exactly the start/end position would create a zero length
			// clip (bad), and a clip the same length as the original one (pointless).
			if (splitPos > m_initialTCOPos && splitPos < m_initialTCOEnd)
			{
				leftTCO->getTrack()->addJournalCheckPoint();
				leftTCO->getTrack()->saveJournallingState(false);

				SampleTCO * rightTCO = new SampleTCO (*leftTCO);

				leftTCO->changeLength(splitPos - m_initialTCOPos);

				rightTCO->movePosition(splitPos);
				rightTCO->changeLength(m_initialTCOEnd - splitPos);
				rightTCO->setStartTimeOffset(leftTCO->startTimeOffset() - leftTCO->length());

				leftTCO->getTrack()->restoreJournallingState();
			}
		}
	}

```

Code style, but more importantly, I think that part of the code would be better extracted to a method inside `SampleTCO`. Then the mouse event would just do something like:
```
SampleTCO * stco = dynamic_cast<SampleTCO*>(m_tco);
if (stco) { stco->splitTCO(me->pos().x()); }
```

Reply:
If it's fine with you I'll extract it, but leave it in `TrackContentObjectView.cpp`. Splitting should really be available for all clips, so eventually all of this code should be shared.

Reply:
Resolved as described in latest commit.